aoc2022/src/main/scala/day12.scala

```diff
@@ -0,0 +1,65 @@
+import scala.io.*
+import math.*
+import scala.annotation.tailrec
+import aoc2022.Algorithms.Dijkstra.*
+import aoc2022.FlatGrid
+
+/**
+ * PART 01:
+ * 
+ * This puzzle was a suitable candidate for my earlier implemented Dijkstra algorithm. I made a function that 
+ * computes the next nodes (type Graph[Int]) based on the index in the map Vector[Char]. 
+ * 
+ * The earlier implemented shortestDistance function was to the rescue here. 
+ *
+ * PART 02:
+ * 
+ * I took all the possible starting positions and then brute forced the answer using the same graph from part 01. 
+ *
+ */
+
+
+object day12 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val input: Vector[String] =
+    Source
+      .fromResource(s"day$day.txt")
+      .getLines
+      .toVector
+
+  def mountainGraph(map: Vector[Char], width: Int): Graph[Int] = 
+    (i: Int) =>
+      val currentHeight: Int = map(i).toInt
+      val neighbours = FlatGrid.neighbours4(i, width, map.length)
+      if neighbours.isEmpty then Map.empty
+      else 
```

Comment:
The `if` statement is redundant here; the implementation below will work fine on an empty list of neighbours I think.